include/cute/atom/copy_traits_xe.hpp

```diff
@@ -38,58 +38,147 @@
+
+template <class T, class dtype, class = void>
+static constexpr auto size_of_inst = sizeof(dtype);
+
+template <class T, class dtype>
+static constexpr auto size_of_inst<T, dtype, std::enable_if_t<has_inst_dtype<T>>> = sizeof(typename T::inst_dtype);
+
+
+// ==========  value_layout_t  ==========
+template <class, class = void>
+static constexpr bool has_dst_shape = false;
+
+template <class T>
+static constexpr bool has_dst_shape<T, cute::void_t<typename T::ValueShape>> = true;
+
+template <class T, class = void>
+struct value_layout_t {
+  using type = decltype(make_layout(make_shape(get<0>(typename T::BlockShape{}),
+                                                            get<1>(typename T::BlockShape{})
+                                                                / Int<detail::subgroup_size>{})));
+};
+
+template <class T>
+struct value_layout_t<T, std::enable_if_t<has_dst_shape<T>>> {
+  using type = decltype(make_layout(make_shape(get<0>(typename T::ValueShape{}),
+                                                            get<1>(typename T::ValueShape{})
+                                                                / Int<detail::subgroup_size>{})));
+};
+
+
+// ==========  is_transpose_load  ==========
+template <class, class = void>
+static constexpr bool has_is_transpose = false;
+
+template <class T>
+static constexpr bool has_is_transpose<T, cute::void_t<decltype(T::is_transpose)>> = true;
 
-  template<>
-  struct is_transpose<XE_2D_U16x16x8_LD_T> : bool_constant<true>{};
+template <class, class = void>
+static constexpr bool is_transpose_load = false;
 
-  template<>
-  struct is_transpose<XE_2D_U16x16x16_LD_T> : bool_constant<true>{};
+template <class T>
+static constexpr bool is_transpose_load<T, std::enable_if_t<has_is_transpose<T>>> = T::is_transpose;
 
-  template<>
-  struct is_transpose<XE_2D_U32x16x2_LD_T> : bool_constant<true>{};
 
-  template<>
-  struct is_transpose<XE_2D_U32x16x4_LD_T> : bool_constant<true>{};
+// ==========  is_stride_leftmost  ==========
+template <class T, class = void>
+static constexpr bool is_stride_leftmost = std::is_same_v<_1, decltype(get<0>(T{}))>;
 
-  template<>
-  struct is_transpose<XE_2D_U32x16x8_LD_T> : bool_constant<true>{};
+template <class T>
+static constexpr bool is_stride_leftmost<T, cute::void_t<decltype(T{}.stride())>> = std::is_same_v<_1, decltype(get<0>(T{}.stride()))>;
 
-  template<>
-  struct is_transpose<XE_2D_U64x8x1_LD_T> : bool_constant<true>{};
 
-  template<>
-  struct is_transpose<XE_2D_U64x8x2_LD_T> : bool_constant<true>{};
+} // end namespace detail
+
+
+template <uint32_t dim, class Tensor_t>
+static constexpr auto append_pvc_tensor(Tensor_t const &t0, uint32_t shape, uint32_t stride) {
+  return make_tensor(make_inttuple_iter(t0.data()), append(t0.layout(), make_layout(shape, E<dim>{} * stride)));
+}
+
+template <class CopyOp, class StrideIndicator = cute::Stride<int64_t, cute::Int<1>, int64_t>>
```

Comment:
Why is the default Stride rank 3 and why is this even allowed to be rank 3? Why is the default stride 1 in the middle?